Transformations/TransformEigen/TransformEigen.cpp

```diff
@@ -0,0 +1,250 @@
+
+        for(int i = 0; i < cv_mat.rows; i++)
+        {
+            for(int j = 0; j < cv_mat.cols; j++)
+            {
+                cv_mat.at<double>(i, j) = eigen_mat(i, j);
+            }
+        }
+
+        return cv_mat;
+    }
+
+    Eigen::MatrixXd cv2eigen(const cv::Mat &cv_mat)
+    {
+        if(cv_mat.dims > 2)
+        {
+            throw "Invalid matrix dimensions. Expected 2D.";
+        }
+
+        Eigen::MatrixXd eigen_mat(cv_mat.rows, cv_mat.cols);
+
+        for(int i = 0; i < cv_mat.rows; i++)
+        {
+            for(int j = 0; j < cv_mat.cols; j++)
+            {
+                eigen_mat(i, j) = cv_mat.at<double>(i, j);
+            }
+        }
+
+        return eigen_mat;
+    }
+
+    Eigen::Vector4d reorderQuaternion(Eigen::Quaterniond &quat_from_rotmat)
+    {
+        Eigen::Vector4d quat_vector_reordered;
+        quat_vector_reordered[0] = quat_from_rotmat.coeffs()[3];
+        for(int i = 0; i < 3; i++)
+            quat_vector_reordered[i + 1] = quat_from_rotmat.coeffs()[i];
+
+        return quat_vector_reordered;
+    }
+
+    //double roll = (-15.016534 * EIGEN_PI) / 180;
+    //double pitch = (4.473777 * EIGEN_PI) / 180;
+    //double yaw = (1.195467 * EIGEN_PI) / 180;
+    //roll = (-15.144396 * EIGEN_PI) / 180;
+    //pitch = (4.009225 * EIGEN_PI) / 180;
+    //yaw = (2.318581 * EIGEN_PI) / 180;
+    Eigen::Array3d rotmat2euler(Eigen::Matrix3d rotmat, std::string orientation = "xyz")
+    {
+        Eigen::Array3d euler_from_rotmat;
+
+        char a[3];
+        for(int i = 0; i < 3; i++)
+        {
+            switch(orientation[i])
+            {
+                case 'x': a[i] = 0; break;
+                case 'y': a[i] = 1; break;
+                case 'z': a[i] = 2; break;
+                default: throw "Invalid orientation";
+            }
+        }
+        euler_from_rotmat = rotmat.eulerAngles(a[0], a[1], a[2]);
+
+        Eigen::Array3d euler360 = euler_from_rotmat * 180 / EIGEN_PI;
+        std::cout << orientation << "(degrees):\n" << euler360 << std::endl;
+        std::cout << orientation << "(radians):\n" << euler_from_rotmat << std::endl;
+
+        return euler_from_rotmat;
+    }
+
+    Eigen::Matrix3d euler2rotmat(Eigen::Array3d euler_from_rotmat, std::string orientation = "xyz")
+    {
+        Eigen::Matrix3d rotmat;
+
+        rotmat = Eigen::Matrix3d::Identity(3, 3);
+        for(int i = 0; i < 3; i++)
+        {
+            switch(orientation[i])
+            {
+                case 'x': rotmat = rotmat * Eigen::AngleAxisd(euler_from_rotmat[i], Eigen::Vector3d::UnitX()); break;
+                case 'y': rotmat = rotmat * Eigen::AngleAxisd(euler_from_rotmat[i], Eigen::Vector3d::UnitY()); break;
+                case 'z': rotmat = rotmat * Eigen::AngleAxisd(euler_from_rotmat[i], Eigen::Vector3d::UnitZ()); break;
+                default: throw "Invalid orientation";
+            }
+        }
+
+        return rotmat;
+    }
+} // namespace TransformUtils
```

Comment:
Why // namespace TransformUtils comment here? Is that a standard way to comment on namespace?

Reply:
I believe I copied that from someone. I'm used to adding a comment to the closing `#endif` of an `#ifdef`. Useful if there are many lines of code in between.

Reply:
clang-format has been [configured](https://github.com/zivid/cpp-extra-samples/blob/master/.clang-format#L51) to add these.